modules/cart/src/Form/AddToCartForm.php

```diff
@@ -81,6 +81,13 @@ public function buildForm(array $form, FormStateInterface $form_state) {
       $form['#access'] = FALSE;
     }
 
+    // If our default variation is set, we alter the add to cart form.
+    if ($default_variation = \Drupal::request()->query->get('dv')) {
+      if (is_numeric($default_variation)) {
+        $form['purchased_entity']['widget'][0]['attributes']['field_variation']['#default_value'] = $default_variation;
```

Comment:
I was thinking about validation as well. But how will we handle it then? Set a message? Should we allow this message to be altered?

I agree on doing this at an earlier stage, but I will need a suggestion where? viewElements?